bson/src/main/org/bson/AbstractBsonWriter.java

```diff
@@ -782,12 +782,26 @@ protected void pipeExtraElements(final List<BsonElement> extraElements) {
         }
     }
 
+    /**
+     * Return true if the current execution of the pipe method should be aborted.
+     *
+     * @return true if the current execution of the pipe method should be aborted.
+     *
+     * @since 3.7
+     */
+    protected boolean abortPipe() {
+        return false;
+    }
+
     private void pipeDocument(final BsonReader reader, final List<BsonElement> extraElements) {
         reader.readStartDocument();
         writeStartDocument();
         while (reader.readBsonType() != BsonType.END_OF_DOCUMENT) {
             writeName(reader.readName());
             pipeValue(reader);
+            if (abortPipe()) {
```

Comment:
This has no visible effect, but makes pipe more efficient by allowing subclasses to abort piping